### ISSUE TITLE
NET-1831 Fix S3900 FN: If parameter is captured, S3900 is not raised

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#7073]
### Description

The LVA removed captured symbols from the SE engine and we raised S3900 as FPs. As a workaround, #7065 was merged. It suppresses S3900 for captured parameters. This creates FNs like this

```cs
public string InsideLambda_CaptureBeforeDereference(string s)
{
    Func<string> someFunc = () => s.ToString();

    return s.ToString(); // FN: s is not checked here
}
```

The `lva.LiveOut()` and/or the calling `RoslynSymbolicExecution.ProcessBranchState()` need to be changed to properly handle captured variables. It is unclear how to solve this best at the moment. Ideally `LiveOut()` would return any symbols access after the current operation by also ignoring any captures. It needs to take the position of the current operation, the capture position, and any other usages into account to do so. Other callees of `LiveOut` may need the current behavior.

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1831](https://sonarsource.atlassian.net/browse/NET-1831)

[NET-1831]: https://sonarsource.atlassian.net/browse/NET-1831?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
